libraries/lib-fft/PffftTransformer.cpp

```diff
@@ -67,22 +102,42 @@ void PffftTransformer::TransformOrdered(PffftConstFloats input,
       assert(false);
       return;
    }
+   const auto N = std::max(MinSize(), requestedSize);
+   if (requestedSize < N) {
+      // In frequency domain, we do an interpolation of the desired transform
+      // of a small window.
+      // This assumes MinSize() is a power of two
+      assert(IsPowerOfTwo(MinSize()));
+      // Zero pad
+      auto pInput = const_cast<float*>(input.get()) + requestedSize;
+      memset(pInput, '\0', (N - requestedSize) * sizeof(float));
+   }
    pffft_transform_ordered(get(), input.get(), output.get(), work.get(),
       PFFFT_FORWARD);
+   if (requestedSize < N) {
+      // Now decimate
+      auto pOutput = output.get();
+      const auto factor = N / requestedSize;
+      for (size_t ii = 1, nn = requestedSize; ii < nn; ++ii)
+         pOutput[ii] = pOutput[ii * factor];
```

Comment:
You overlooked that `output` here has layout `[DC, Nyq, real1, imag1, ...]`.
But besides that I think it should work.